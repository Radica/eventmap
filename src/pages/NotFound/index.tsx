import React, { memo } from 'react';
import { Helmet } from 'react-helmet';

import styles from './styles.scss';

type Props = {
    staticContext?: { status: string };
};

export default memo(({ staticContext }: Props) => {
    // We have to check if staticContext exists
    // because it will be undefined if rendered through a BrowserRoute
    /* istanbul ignore next */
    if (staticContext) staticContext.status = '404';

    return (
        <div className={styles.NotFound}>
            <Helmet title="Not Found" />
            <p>Page was not found!</p>
        </div>
    );
});
