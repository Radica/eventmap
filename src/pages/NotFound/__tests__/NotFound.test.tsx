import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import NotFound from '../index';

describe('<NotFound />', () => {
    it('renders', () => {
        const tree = render(
            <MemoryRouter>
                <NotFound />
            </MemoryRouter>
        ).container.firstChild;

        expect(tree).toMatchSnapshot();
    });
});
