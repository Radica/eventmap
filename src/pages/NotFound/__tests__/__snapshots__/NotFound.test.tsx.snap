// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders 1`] = `
<div
  class="NotFound"
>
  <p>
    Oops, Page was not found!
  </p>
</div>
`;
